pkgs/development/python-modules/crytic-compile/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, buildPythonPackage, fetchPypi, makeWrapper, pythonOlder
+, setuptools
+, pysha3
+, solc
+}:
+
+buildPythonPackage rec {
+  pname = "crytic-compile";
+  version = "0.1.6";
+
+  disabled = pythonOlder "3.6";
+
+  # No Python tests
+  doCheck = false;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "32c30a9c4e6d2122fe01b21648127bca949c43402ed5a91c10bf3aa34a53731a";
+  };
+  
+  nativeBuildInputs = [ makeWrapper ];
+  propagatedBuildInputs = [ pysha3 setuptools ];
+
+  postFixup = ''
+    wrapProgram $out/bin/crytic-compile \
+      --prefix PATH : "${lib.makeBinPath [ solc ]}"
+  '';
```

Comment:
does the package also shell out to solc? if it does, then this wont work if someone imports this as a package.

Reply:
If used as a standalone tool, yes. 
I am not aware of any nix package that depends on this besides `slither-analyzer`, which also includes this `postFixup`.

Reply:
> does the package also shell out to solc? if it does, then this wont work if someone imports this as a package.

@jonringer why is that?

Reply:
because you only wrap the executable, and the python package package has no knowledge of it.

If i had another package which has `propagatedBuildInputs = [ cryptic-compile ];` as a dependenccy, then when I call `$ my_package_exe` and it eventually calls cryptic-compile, which will do something to the effect of `subprocess.check_call("solc" ...)`, then solc won't be present.

If you want the package to do the right thing, you need to patch the call sights inside of only wrapping the entry_points.